src/test/java/io/lakefs/routerfs/RouterFileSystemTest.java

```diff
@@ -0,0 +1,143 @@
+package io.lakefs.routerfs;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.*;
+import org.apache.hadoop.fs.permission.FsPermission;
+import org.apache.hadoop.util.Progressable;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.io.IOException;
+import java.net.URI;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+public class RouterFileSystemTest {
+    private static final String DEFAULT_FS_CONF_KEY_FORMAT = "routerfs.default.fs.%s";
+    private static final String DEFAULT_FS_SCHEME = "def";
+    private static final Path PATH = new Path("some://path/");
+    private static final URI uri = URI.create("scheme://authority");
+
+    private RouterFileSystem routerFileSystemUnderTest;
+
+    @Mock private PathMapper mockPathMapper;
+    @Mock private Path mockPath;
+    @Mock private FileSystem mockFileSystem;
+
+    @BeforeEach
+    private void setUp() throws IOException {
+        lenient().when(this.mockPathMapper.mapPath(any())).thenReturn(this.mockPath);
+        this.routerFileSystemUnderTest = new RouterFileSystem(this.mockPathMapper);
+        Configuration configuration = generateConfiguration();
+        lenient().when(this.mockPath.getFileSystem(configuration)).thenReturn(this.mockFileSystem);
+        this.routerFileSystemUnderTest.initialize(uri, configuration);
+    }
+
+    @Test
+    public void testGetUriSuccessful() {
+        URI uri = routerFileSystemUnderTest.getUri();
+        assertEquals(uri.getPath(), uri.getPath());
```

Comment:
Is this test a WIP? we are comparing an object to iteself

Reply:
😅​ it was but then just forgot to delete it...
Thanks